### PR TITLE
Implement Industry::expandGrounds

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Ready for review. I hooked this function so it would be spammed a lot and i am getting no divergence on the singular industry save.

### CODE REVIEW

File: src/OpenLoco/Industry.cpp [hunk 1]
```diff
@@ -439,30 +440,183 @@ namespace OpenLoco
...
+        const auto* indObj = ObjectManager::get<IndustryObject>(objectId);
+        struct UnkStruct
+        {
+            Utility::prng prng;
+            uint8_t varEC;
+        };
+        std::optional<UnkStruct> E0C3DC;
+        if (indObj->flags & IndustryObjectFlags::unk23)
+        {
+            E0C3DC = UnkStruct{ prng, indObj->var_EC };
```
Comment (duncanspumpkin) [inline src/OpenLoco/Industry.cpp:548]: I've struggled to test this bit of code here.

File: src/OpenLoco/Industry.cpp [hunk 2]
```diff
@@ -439,30 +440,183 @@ namespace OpenLoco
...
+            if (E0C3DC.has_value())
+            {
+                const auto randVal = E0C3DC->prng.randNext();
+                dl = (((randVal & 0xFF) * E0C3DC->varEC) / 256)
+                    | (((randVal >> 8) & 0x7) << 5);
+            }
+            bool skipBorderClear = false;
+            if (is27)
+            {
+                if (!E0C3DC.has_value()) // Vanilla mistake here check to see if anyone suffers from it
```
Comment (duncanspumpkin) [inline src/OpenLoco/Industry.cpp:573]: and here

File: src/OpenLoco/Industry.cpp [hunk 3]
```diff
@@ -439,30 +440,183 @@ namespace OpenLoco
...
+            randWallTypeFlags |= 1 << ((prng.srand_0() >> 4) & 0x1F);
+        }
+
+        std::size_t i = 0;
+        for (const auto& tilePos : TilePosRangeView{ topRight, bottomLeft })
+        {
+            if (E0C3DC.has_value())
+            {
+                const auto randVal = E0C3DC->prng.randNext();
+                dl = (((randVal & 0xFF) * E0C3DC->varEC) / 256)
```
Comment (duncanspumpkin) [inline src/OpenLoco/Industry.cpp:567]: and basically here

File: src/OpenLoco/Industry.cpp [hunk 4]
```diff
@@ -439,30 +440,179 @@ namespace OpenLoco
...
+        std::optional<Utility::prng> is23prng;
+        if (indObj->flags & IndustryObjectFlags::unk23) // Livestock use this
+        {
+            is23prng = prng;
+        }
+        std::optional<Utility::prng> is27prng;
+        if (indObj->flags & IndustryObjectFlags::unk27) // Skislope use this
+        {
+            // Vanilla mistake here didn't set the prng! It would just recycle from a previous unk23 caller
+            is27prng = prng;
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Industry.cpp:551]: Any notable divergence as a result of this change?

Comment (duncanspumpkin) [reply src/OpenLoco/Industry.cpp:551]: Yes there would be but I've not really got a good setup to show the difference as its only used by ski centres.

**Pull request merged**